crates/goose/src/providers/formats/anthropic.rs

```diff
@@ -96,18 +96,14 @@ pub fn format_messages(messages: &[Message]) -> Vec<Value> {
                 MessageContent::SummarizationRequested(_) => {
                     // Skip
                 }
-                MessageContent::Thinking(thinking) => {
-                    content.push(json!({
-                        TYPE_FIELD: THINKING_TYPE,
-                        THINKING_TYPE: thinking.thinking,
-                        SIGNATURE_FIELD: thinking.signature
-                    }));
+                MessageContent::Thinking(_) => {
+                    // Skip thinking blocks - they should not be sent to the API
+                    // Thinking is only for internal use
+                    continue;
                 }
-                MessageContent::RedactedThinking(redacted) => {
-                    content.push(json!({
-                        TYPE_FIELD: REDACTED_THINKING_TYPE,
-                        DATA_FIELD: redacted.data
-                    }));
+                MessageContent::RedactedThinking(_) => {
+                    // Skip redacted thinking - internal use only
+                    continue;
```

Comment:
I'm not sure about this. it sounds reasonable, but do we want to make a change we don't really really understand for this if we want to cherry-pick?